### PR TITLE
cleanup(pubsub)!: remove concurrency watermarks

### LINKED ISSUE #5388: Simplify PublisherOptions concurrency control
No LWM and HWM settings, just a max.

### PR DESCRIPTION
**Pull request by coryan**
**BREAKING CHANGE:**

* Simplify the concurrency control configuration in `pubsub::SubscriberOptions`,
  applications only need to set the maximum number of messages that will be
  scheduled in parallel.

Fixes #5388

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5389)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,10 @@
 While the Pub/Sub library is not GA, and breaking changes are to be expected, we
 are close enough to a GA release that we think highlighting them is important.
 
+* Simplify the concurrency control configuration in `pubsub::SubscriberOptions`,
+  applications only need to set the maximum number of messages that will be
```
Comment (devbww) [inline CHANGELOG.md:29]: s/, applications/. Applications/

File: google/cloud/pubsub/internal/subscription_concurrency_control_test.cc [hunk 2]
```diff
@@ -221,10 +222,10 @@ TEST_F(SubscriptionConcurrencyControlTest, ParallelCallbacks) {
   EXPECT_THAT(done.get(), StatusIs(StatusCode::kOk));
 }
 
-/// @test Verify SubscriptionConcurrencyControl respects the HWM limit.
-TEST_F(SubscriptionConcurrencyControlTest, ParallelCallbacksRespectHwmLimit) {
-  auto constexpr kConcurrencyHwm = 8;
-  auto constexpr kConcurrencyLwm = 0;
+/// @test Verify SubscriptionConcurrencyControl respects the concurrency limit.
+TEST_F(SubscriptionConcurrencyControlTest,
+       ParallelCallbacksRespectConcurrencyimit) {
```
Comment (devbww) [inline google/cloud/pubsub/internal/subscription_concurrency_control_test.cc:227]: s/Concurrencyimit/ConcurrencyLimit/

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/internal/subscription_concurrency_control_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/pubsub/internal/subscription_concurrency_control_test.cc]: Thanks, fixed..

File: CHANGELOG.md (hunk 1 above)
Comment (coryan) [reply CHANGELOG.md]: Done.

**Pull request merged**